simulator/src/resim/cmd_set_default_account.rs

```diff
@@ -19,12 +19,47 @@ pub struct SetDefaultAccount {
 impl SetDefaultAccount {
     pub fn run<O: std::io::Write>(&self, out: &mut O) -> Result<(), Error> {
         let mut configs = get_configs()?;
+        let private_key = parse_private_key_from_str(&self.private_key).map_err(|e| {
+            if Secp256k1PublicKey::from_str(&self.private_key).is_ok() {
+                Error::GotPublicKeyExpectedPrivateKey
```

Comment:
If we fail to create a PrivateKey (32 bytes/64 hex char input) from the passed in string, we try to parse it as a PublicKey to see if the user made the very reasonable error of accidentally copy pasting the public key - like I did!